lib/modules/manager/osgi/extract.ts

```diff
@@ -0,0 +1,128 @@
+import is from '@sindresorhus/is';
+import * as json5 from 'json5';
+import { coerce, satisfies } from 'semver';
+import { logger } from '../../../logger';
+import { MavenDatasource } from '../../datasource/maven';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+import type { Bundle, FeatureModel } from './types';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config?: ExtractConfig
+): PackageFile | null {
+  // References:
+  // - OSGi compendium release 8 ( https://docs.osgi.org/specification/osgi.cmpn/8.0.0/service.feature.html )
+  // - The Sling implementation of the feature model ( https://sling.apache.org/documentation/development/feature-model.html )
+  logger.debug({ fileName }, 'osgi.extractPackageFile');
+
+  const deps: PackageDependency[] = [];
+  let featureModel: FeatureModel;
+  try {
+    // Compendium R8 159.3: JS comments are supported
+    featureModel = json5.parse<FeatureModel>(content);
+  } catch (e) {
+    logger.warn(`Failed parsing ${fileName}: ${(e as Error).message}`);
+    return null;
+  }
+
+  // Compendium R8 159.9: resource versioning
+  if (!isSupportedFeatureResourceVersion(featureModel, fileName)) {
```

Comment:
featureModel can be null here when the json is `null` which is valid json5, so validate with `is.plainObject` too.

Reply:
I had to revert this change in 5a0108222 as it was making the linter unhappy. I am not sure exactly what is going on, but it seems that a check like

```javascript
if (!is.plainObject(featureModel)) {
  return null
}
```

or even 
```javascript
if (!is.plainObject<FeatureModel>(featureModel)) {
  return null
}
```

affects the effective type of the `featureModel` variable. I'm open to suggestions on how to fix this in a more elegant manner, but for now I simply check for null and undefined.